### PR TITLE
BUGZ-1321: Address race condition on accessing a resource of the OpenGLDisplayPlugin before it s garanteed to be allocated

### PR DESCRIPTION
**Pull request by samcake**
https://highfidelity.atlassian.net/browse/BUGZ-1321

This crash is caused when accessing the size of the _compositeBuffer of the OpenGLDisplayPlugin from the render thread. Sometime, that resource is not allocated.
I replaced the the _compositeBuffer->getSize() by getRecommendedREnderSize() which is returning just value member.
The resource _compositeBuffer is allocated on the first present loop call setting up the display plugin and this can happen AFTER the render Thread starts using it (in the case of getting its size).

As a follow up, i removed the need for the stereo info about the eyeViewport where this can be avoided in OpenGLDIsplayPLugin.
I moved the function and the only need for it in HMDDisplayPlugin.

TEST PLAN:
In HMD mode (all the hmd vendors)
- Validate that the mouse cursor is displayed correctly in stereo
- validate that the overlay ui (stats for developers) is displayed correctly in stereo
In Desktop, nothing should have changed

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21225/HighFidelity-Beta-PR16161-a290723.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21225/HighFidelity-Quest-Beta-PR16161-a290723.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21225/HighFidelity-Beta-PR16161-a290723.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21225/HighFidelity-Beta-Interface-PR16161-a290723.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21225/HighFidelity-Beta-PR16161-a290723.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21225/HighFidelity-Beta-Interface-PR16161-a290723.exe

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.h [hunk 1]
```diff
@@ -34,6 +34,8 @@ class HmdDisplayPlugin : public OpenGLDisplayPlugin {
     glm::uvec2 getRecommendedRenderSize() const override final { return _renderTargetSize; }
```
Comment (AndrewMeadows) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.h]: This code in this PR looks fine but I noticed: `_renderTargetSize` is a `glm::uvec2` and is never explicitly set by the ctor or any base init code.  (Which means it is using whatever glm initializes it to, which is probably `<0,0>`, and which could go away in future **glm** versions, but that is besides the current point.) Instead it is set by some (but not all!) derived plugins:

```$ find ./ -type f -name "*.cpp" -exec grep -Hn _renderTargetSize {} \; | grep "_renderTargetSize ="
./plugins/oculusLegacy/src/OculusLegacyDisplayPlugin.cpp:173:    _renderTargetSize = uvec2(
./plugins/oculus/src/OculusBaseDisplayPlugin.cpp:148:    _renderTargetSize = uvec2(
./libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp:60:    _renderTargetSize = { 1024, 512 };
./libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp:65:        _renderTargetSize = glm::uvec2{
./libraries/display-plugins/src/display-plugins/hmd/DebugHmdDisplayPlugin.cpp:73:    _renderTargetSize = uvec2{ 1214 * 2 , 1344 };
```
I find it odd that `plugins/openvr/src/OpenVrDisplayPlugin.cpp` never sets it.  Does it not need it?

**Pull request merged**